content/migration.md

```diff
@@ -68,6 +68,176 @@ In the version 6 major release of NestJS, we introduced the code-first approach
+      url: '/health',
+      healthIndicators: [
+        async () => this.dns.pingCheck('google', 'https://google.com'),
+      ],
+    };
+    return {
+      endpoints: [healthEndpoint],
+    };
+  }
+}
+
+@Module({
+  imports: [
+    TerminusModule.forRootAsync({
+      useClass: TerminusOptionsService
+    })
+  ]
+})
+export class AppModule { }
+
+// After
+@Controller('health')
+export class HealthController {
+  constructor(
+    private health: HealthCheckService,
+    private dns: DNSHealthIndicator,
+  ) { }
+
+  @Get()
+  @HealthCheck()
+  healthCheck() {
+    return this.health.check([
+      async () => this.dns.pingCheck('google', 'https://google.com'),
+    ]);
+  }
+}
+
+@Module({
+  imports: [
+    TerminusModule
+  ]
+})
+export class AppModule { }
+
+@@switch
+
+// Before
+@Injectable()
+@Dependencies(DNSHealthIndicator)
+export class TerminusOptionsService {
+  constructor(
+    private dns,
+  ) {}
+
+  createTerminusOptions() {
+    const healthEndpoint = {
+      url: '/health',
+      healthIndicators: [
+        async () => this.dns.pingCheck('google', 'https://google.com'),
+      ],
+    };
+    return {
+      endpoints: [healthEndpoint],
+    };
+  }
+}
+
+@Module({
+  imports: [
+    TerminusModule.forRootAsync({
+      useClass: TerminusOptionsService
+    })
+  ]
+})
+export class AppModule { }
+
+// After
+@Controller('/health')
+@Dependencies(HealthCheckService, DNSHealthIndicator)
+export class HealthController {
+  constructor(
+    private health,
+    private dns,
+  ) { }
+
+  @Get('/')
+  @HealthCheck()
+  healthCheck() {
+    return this.health.check([
+      async () => this.dns.pingCheck('google', 'https://google.com'),
+    ])
+  }
+}
+
+@Module({
+  controllers: [
+    HealthController
+  ],
+  imports: [
+    TerminusModule
+  ]
+})
+export class AppModule { }
+```
+
+> warning **Warning** In case you have set a [Global Prefix](faq#global-prefix) in your Nest application and you have not used the `useGlobalPrefix` Terminus option, the URL of your health check will change! Make sure to update the reference to that URL, or use the legacy Terminus API until [nestjs/nest#963](https://github.com/nestjs/nest/issues/963) is fixed.
+
+In case there is a problem with the new NestJS Terminus API, make sure you report it on [the repository](https://github.com/nestjs/terminus/issues). In case you are forced to use the legacy API, you can also disable deprecation messages for the time being.
+
+```typescript
+TerminusModule.forRootAsync({
+  useFactory: () => ({
+    disableDeprecationWarnings: true,
+    endpoints: [
+      // ...
+    ]
+  })
+}
+```
+
+Besides, it is recommended to enable shutdown hooks in your `main.ts`. The Terminus integration
+will listen on [POSIX signals](https://en.wikipedia.org/wiki/Signal_(IPC)#POSIX_signals) such as `SIGTERM`.
+
+To enable this, checkout the [Application shutdown docs](fundamentals/lifecycle-events#application-shutdown).
```

Comment:
Suggestion: combine/modify to a single paragraph:

"You should enable shutdown hooks in your main.ts. The Terminus integration will listen on POSIX signals such as SIGTERM.  See the  [Application shutdown chapter](fundamentals/lifecycle-events#application-shutdown) for more information.  When enabled, the health check route(s) will automatically respond with a Service Unavailable (503) HTTP error response when the server is shutting down.

Reply:
Thanks, resolved! <3